### PR TITLE
Enable the filesystem preferences by splitting the preferences extension

### PR DESCRIPTION
**Pull request by akosyakov**

### CODE REVIEW

**Comment by akosyakov** [comment]
if nobody minds I am going to merge it tomorrow morning.

File: packages/preferences/src/common/index.ts [hunk 1]
```diff
@@ -5,9 +5,5 @@
  * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
  */
 
-export * from './preference-protocol';
-export * from './preference-service';
-export * from './preference-proxy';
-export * from './compound-preference-server';
-export * from './default-preference-server';
+export * from '@theia/preferences-api';
```
Comment (svenefftinge) [inline packages/preferences/src/common/index.ts]: why does it need to re-export everything?

Comment (akosyakov) [inline packages/preferences/src/common/index.ts]: did not want to refactor existing imports

**Comment by epatpol** [comment]
I'm good with this! I'll be waiting for this one to be merged to rebase my PR on master.

File: packages/preferences/src/common/index.ts (hunk 1 above)
Comment (svenefftinge) [inline packages/preferences/src/common/index.ts]: I think the imports should be `@theia/preferences-api/lib/common`

Comment (svenefftinge) [inline packages/preferences/src/common/index.ts]: nevermind, I see you adjusted the default in the generator (I think that should be explicit in the extension-package.json)

**Pull request merged**